### DISCUSSION TITLE
Domain verification error

### DISCUSSION THREAD

**Discussion by MickaelNeves** [WalletConnect/walletconnect-monorepo#3343] [Developer Support] [answered]
I added the TXT to vercel but still getting error when trying to validate domain
![Screenshot 2023-08-02 at 12 49 24](https://github.com/WalletConnect/walletconnect-monorepo/assets/20476271/986c4246-c033-4f4f-ab7e-4529f6490cd4)
![Screenshot 2023-08-02 at 12 38 33](https://github.com/WalletConnect/walletconnect-monorepo/assets/20476271/c5c5b8db-f316-4408-b3e9-8920a1dabeef)

**Comment by ligi** [comment]
seems you TXT record is not properly added

```
$ dig botto.com TXT

; <<>> DiG 9.18.12-1ubuntu1.1-Ubuntu <<>> botto.com TXT
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 42397
;; flags: qr rd ra; QUERY: 1, ANSWER: 2, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 65494
;; QUESTION SECTION:
;botto.com.			IN	TXT

;; ANSWER SECTION:
botto.com.		273	IN	TXT	"google-site-verification=6U_xckvGgNAs8e3cRkDd5tmjZvx8BdQy_FSUISqt61k"
botto.com.		273	IN	TXT	"google-site-verification=BGXp8pxyH-U2m-8d4nOduLWVkByoGYuY4iOLm5Tygeg"

;; Query time: 28 msec
;; SERVER: 127.0.0.53#53(127.0.0.53) (UDP)
;; WHEN: Wed Aug 02 12:03:37 CEST 2023
;; MSG SIZE  rcvd: 200

```

**Comment by boidushya** [comment, marked as answer]
seems like youve added the flag for "wallet-connect-dns-verficiation" subdomain. Instead, you need to add it for your main domain, and copy-paste the string exactly how it shows up while keeping the "Name" section empty
A correct configuration would show up like this on dig
```
boidushya.com.		600	IN	TXT	"wallet-connect-dns-verification=b7e0...9be1"
```

**Reply by MickaelNeves** [reply]
yap that was the issue, thanks!

**Comment by michallangmajer** [comment]
Hello, I'm facing a similar problem with coinbrain.com. How can I validate whether the TXT record was correctly added? 😞 From my POV everything looks fine. 🙏 Any ideas, please?

**Comment by boidushya** [comment]
> Hello, I'm facing a similar problem with coinbrain.com. How can I validate whether the TXT record was correctly added? 😞 From my POV everything looks fine. 🙏 Any ideas, please?

Hey, can you try using mxtoolbox.com/txtlookup.aspx for this?

**Comment by cryptojoindev** [comment]
I vahe added TXT correctally but not work
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/142065259/ae74a109-930c-40ff-a60c-6b52dc9dee90)

![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/142065259/21c7b6f2-6423-4145-8cce-4952fb617864)

**Reply by boidushya** [reply]
Looks like you might've set up your domain incorrectly on your WalletConnect Cloud dashboard, since I see you're using `com.br` instead of `cryptojoin.com.br`

**Reply by cryptojoindev** [reply]
Domain is correct 
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/142065259/042ead91-3f03-4b8d-956e-9728d51a7819)

**Reply by cryptojoindev** [reply]
In domain verification always show ".com.br", I think that is a bug in backend of wallet connect

**Reply by Cali93** [reply]
Hey @cryptojoindev, thanks for the report.
The issue has ben fixed, you should be able to head over https://cloud.walletconnect.com and copy the newly generated verification code for `cryptojoin.com.br` instead of `com.br`. You will then need to update your TXT record to point to the new verification code.

Sorry for the inconvenience, let us know if it worked out or if you're still facing issues.

**Comment by michallangmajer** [comment]
In our case, the problem was that we set everything correctly only in GoDaddy administration but we're using CloudFlare for CDN and it was necessary to setup there as well.

**Comment by ravisriv** [comment]
Domain verification issue again on Walletconnect.com when verifying domain at https://cloud.walletconnect.com/app as of September 15, 2023. 

I had my domain verified earlier. Recently, I made a minor change and tried to re-verify the domain but it no longer works. I tried all the different permutations and combinations. The key of the TXT record has changed. Instead of "wallet-connect-dns-verification," the key is now a hex string. I tried replacing the key with the old standard ("wallet-connect-dns-verification"). That did not work. I tried the key the way it is - straight copy-paste. That did not work too.

![Screenshot-walletconnect](https://github.com/WalletConnect/walletconnect-monorepo/assets/62220449/05e6767b-3821-4729-bd61-72e9d26fb012)

Here is the error I get when I copy-paste the verification code for the TXT record.
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/62220449/3996b297-f900-4561-8cc3-b6c8cfb23b12)

And checked on mxtoolbox.com/txtlookup.aspx. The DNS is publishing the TXT record correctly. The issue appears to be with the verification program of Walletconnect. It is not matching the key correctly.

**Comment by oxshaman** [comment]
I am having the same issue! Tried verifying it in 10 different ways, recreating keys, etc... and nothing works. It seems that the verify process is now broken.

**Reply by ravisriv** [reply]
Yes, this is an issue. Verify process is broken. It is not working if you want to re-verify a domain that you had previously verified. There is a workaround though. Unfortunately, the workaround is not pretty.

Since the issue is with the re-verification of a previously verified domain, you have to scroll down on the Walletconnect dashboard and delete your entire project.

Next set up a new project with the same domain. Now you'll be able to verify it.

The pain in the xyz is that you'll have to change the project id, project secrets, etc. in your software that calls Walletconnect.

Let me know if this works for you.

**Reply by albymack** [reply]
I have a newly created project that also doesn't work when trying to verify.  Same error as everybody else.  This is not a re-verify of an previously verified project.  This is a newly created project.

Using mxtoolbox.com/txtlookup.aspx, the txt record is showing up as it should.

**Reply by ravisriv** [reply]
@albymack - Delete the project. After that re-create the project by clicking on the "New Project" button and completing the profile. I had tested this procedure on a domain yesterday and it had worked. But the process is continuing to fail for previously verified domains.

**Reply by Cali93** [reply]
There are only 2 ways of verifying your domain. Please follow our detailed guide https://docs.walletconnect.com/2.0/cloud/verify and give it some time (it can take up to 72 hours for your DNS record to propagate) also make sure that the TTL of previous/stale records isn't too long.

**Reply by oxshaman** [reply]
@Cali93 - the DNS has been propagated I used `dig` to check the DNS records. I have followed the guide to the letter.

This is the response for `dig klaster.io TXT`
```
; <<>> DiG 9.10.6 <<>> klaster.io TXT
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 12334
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
;; QUESTION SECTION:
;klaster.io.			IN	TXT

;; ANSWER SECTION:
klaster.io.		3600	IN	TXT	"3751743d-1dfc-41b2-b002-071d1be358bb=hidden-info-hidden-info"

;; Query time: 159 msec
;; SERVER: 192.168.1.1#53(192.168.1.1)
;; WHEN: Wed Sep 20 11:17:33 CEST 2023
;; MSG SIZE  rcvd: 153
``` 

Yet, when I click verify on the site it gets broken.

I'm quite sure you guys have a bug in the verification process.
<img width="577" alt="Screenshot 2023-09-20 at 11 18 24" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/129866940/ca2dff35-c8d0-4609-9140-157d1e6e16d9">

**Reply by Cali93** [reply]
Thanks for the additional context and sorry for the inconvenience.
The domain verification was relying on a 200 status code to check if you used the static file method.
And in your case `curl --location 'https://klaster.io/.well-known/walletconnect.txt'` is returning 200.
This is a valid edge case and the issue has been fixed.

**Reply by oxshaman** [reply]
Oh amazing! This is probably partially my fault also, but the speed of this fix is impressive. Thanks guys 👍

**Comment by Ductm00** [comment]
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/117615538/c71883e6-9a21-4918-99da-478051ba24b6)
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/117615538/17ec9be2-2eaf-4fd9-a588-ce4bbad2033b)
This is my curl result but not verified domain

**Reply by Ductm00** [reply]
@Cali93 help me please

**Reply by darrenkel** [reply]
Any update on this?

**Reply by Cali93** [reply]
Hey there, this issue has been fixed since then. If you're facing problems verifying your domain please provide information so that we can help you.

**Reply by haiifeng** [reply]
me too, help.

**Reply by c0da523** [reply]
Is it only support second level domain?

**Reply by Cali93** [reply]
Yes, once you verified your second level domain it should work for all subdomains.

**Comment by ketan-esp** [comment]
Domain Verification failing.
![Screenshot 2024-10-23 at 6 32 25 PM](https://github.com/user-attachments/assets/5238010e-03d6-458b-9559-8bcdefcf53e9)
![Screenshot 2024-10-23 at 6 32 38 PM](https://github.com/user-attachments/assets/cfdb2ae2-3a82-42ef-893a-da086a53f8c0)
![Screenshot 2024-10-23 at 6 33 02 PM](https://github.com/user-attachments/assets/b4e772ae-4601-4e5a-ab9f-fa054d6c0a8a)
